src/App.ts

```diff
@@ -314,6 +322,27 @@ export default class App {
     );
   }
 
+  public shortcut<Shortcut extends SlackShortcut = SlackShortcut>(
+    callbackId: string | RegExp,
+    ...listeners: Middleware<SlackShortcutMiddlewareArgs<Shortcut>>[]
+  ): void;
+  public shortcut<Shortcut extends SlackShortcut = SlackShortcut>(
+    constraints: ShortcutConstraints,
```

Comment:
this is great for simplified cases, but we might be able to do a little better. similar to the `.action()` method, if the `ShortcutConstraints` can deal with a generic type parameter, then we can get more arrow types in the listener. for example, using `{ type: 'shortcut', callback_id: 'cid' }` as the constraints could help type the `shortcut` argument as a `GlobalShortcut` instead of as a `GlobalShortcut | MessageShortcut`.